Gems/AWSClientAuth/Code/Source/AWSClientAuthSystemComponent.cpp

```diff
@@ -169,10 +169,29 @@ namespace AWSClientAuth
         AZ::Interface<IAWSClientAuthRequests>::Register(this);
         AWSClientAuthRequestBus::Handler::BusConnect();
 
-        // Objects below depend on bus above.
         m_authenticationProviderManager = AZStd::make_unique<AuthenticationProviderManager>();
-        m_awsCognitoUserManagementController = AZStd::make_unique<AWSCognitoUserManagementController>();
-        m_awsCognitoAuthorizationController = AZStd::make_unique<AWSCognitoAuthorizationController>();
+
+        // Sanity check if code should setup Cognito user and autorization controllers.
+        // Only set up if Cognito settings appear to be provided in resource mapping file.
+        AZStd::string userPoolId;
+        AWSCore::AWSResourceMappingRequestBus::BroadcastResult(
+            userPoolId, &AWSCore::AWSResourceMappingRequests::GetResourceNameId, CognitoUserPoolIdResourceMappingKey);
+
+        AZStd::string cognitoIdentityPoolId;
+         AWSCore::AWSResourceMappingRequestBus::BroadcastResult(
+            cognitoIdentityPoolId, &AWSCore::AWSResourceMappingRequests::GetResourceNameId, CognitoIdentityPoolIdResourceMappingKey);
+
+        if (userPoolId.empty() && cognitoIdentityPoolId.empty())
```

Comment:
Do we support any uses cases which require a "user pool" but _not_ an "identity pool"? Say, if the project requires user sign-in and manages user profiles, but doesn't otherwise interact with other AWS services (which would require exchanging tokens with an identity pool)? Wondering if there's a situation where we'd want to initialize the `AWSCognitoUserManagementController` but not the auth controller.